docs/tutorials/cpp.md

```diff
@@ -1,43 +1,75 @@
+Typically C++ users can build ICU4X by building the `icu_capi` Rust crate, and linking the resultant static library to their C++ application. This crate contains all of the relevant [Diplomat]-generated `extern "C"` declarations.
 
 Using ICU4X in C++ is best demonstrated via the [examples] present in the codebase. For example, [here's an example showing off decimal formatting in ICU4X][decimal-example-code], built with [this Makefile][decimal-example-makefile].
 
 _We are still working on improving the user experience of using ICU4X from other languages. As such, this tutorial may be a bit sparse, but we are happy to answer questions on our [discussions forum] and help you out_
 
-## Building and linking ICU4X
-
-- `cargo build` only supports local crates, so you have to download the published code from crates.io
-    - You can use `cargo clone icu_capi_staticlib@1.0.0` to do this (`cargo install cargo-clone` if the command is not installed)
-- Inside the crate you can use `cargo build` to build the library
-    - Be sure to pass `--release` to get an optimized build
-    - Specify Cargo features with `--features ...`:
-        - `compiled_data` \[default\] to include data (`ICU4XDataProvider::create_compiled()`)
-        - `simple_logger` \[default\] enable basic stdout logging of error metadata. Further loggers can be added on request.
-        - `buffer_provider` for working with blob data providers (`ICU4XDataProvider::create_from_byte_slice()`)
-        - `provider_fs` for loading data from the file system (`ICU4XDataProvider::create_fs()`). This also requires enabling a syntax on the `icu_provider` crate.
-    - Set `CARGO_PROFILE_RELEASE_LTO=true` to enable link-time optimization
-    - Set `CARGO_PROFILE_RELEASE_OPT_LEVEL="s"` to optimize for size
-    - See [cargo profiles](cargo-profiles) for more options
- - Copy the header files from the `icu_capi` crate
-    - If you're using the default crate registry, they will be in `~/.cargo/registry/src/*/icu_capi-1.0.0/cpp/include`
-    - If you cannot find the crate, download it from crates.io
-    - Make sure to use the same version as above
-        - While we selected a version for `icu_capi_staticlib` before, and here we're looking at `icu_capi`, those versions will always be in sync. You can sanity check this by running `cargo pkgid -p icu_capi`.
- - Compile with `g++ -std=c++17 icu/target/release/libicu_capi_staticlib.a -ldl -lpthread -lm`. C++ versions beyond C++17 are supported, as are other C++ compilers.
+## Building ICU4X
+
+After [installing Rust](https://rustup.rs), create a local Rust crate to configure ICU4X:
+
+```shell
+cargo init icu --lib
+cd icu
+echo "[lib]" >> Cargo.toml
+echo "crate-type = ['cdylib', 'staticlib']" >> Cargo.toml
```

Comment:
Nit, here and elsewhere: rather than relying on people knowing how `echo` works, I'd prefer just giving them text to copy and paste into the file.

Thinking of the tutorial we gave at UTW, if the first instruction was "run echo in your terminal", we may have lost a large fraction of the crowd.

Reply:
I think we can expect that a C++ developer knows how echo works. I'd much rather copy some commands to my terminal than copy two commands, then open a text editor, muck around with it, and then go back to the terminal.

Also by staying on the command line this can be put into a make file.

Reply:
done